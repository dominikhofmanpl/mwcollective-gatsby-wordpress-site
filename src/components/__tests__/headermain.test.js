import * as React from 'react'
import testphoto from '../../assets/fonts/break.jpg'
import renderer from "react-test-renderer"

import HeaderMain from "../headermain"
import { StaticQuery } from 'gatsby'

beforeEach(() => {
  StaticQuery.mockImplementationOnce(({ render }) =>
    render({
      pageBy : {
        subpageFields : {
          subpageCover : {
              sourceUrl
          },
            subpageTitle,
            subpageDesc,
            callToActionName,
            callToActionSlug
          }
      }
    })
  )
})
describe("Index", () => {
  it("renders correctly", () => {
    const data = {
      pageBy : {
        subpageFields : {
          subpageCover : {
              sourceUrl: `${testphoto}`
          },
            subpageTitle: 'mniej znaczy więcej',
            subpageDesc: 'testowy opis',
            callToActionName: 'czytaj więcej',
            callToActionSlug: '/czytaj'
          }
      }
    }
    const tree = renderer.create(<HeaderMain 
          HeaderImage={data.pageBy.subpageFields.subpageCover}
          pageTitle={data.pageBy.subpageFields.subpageTitle}
          pageDescription={data.pageBy.subpageFields.subpageDesc}
          height="70vh">
        </HeaderMain>).toJSON()
    expect(tree).toMatchSnapshot()
  })
})