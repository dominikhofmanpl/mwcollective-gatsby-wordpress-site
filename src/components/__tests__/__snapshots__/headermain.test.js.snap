// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index renders correctly 1`] = `
<header
  className="grid sm:grid-cols-1 lg:grid-cols-2 pt-32 md:pt-0 px-8 md:px-0"
  style={
    Object {
      "minHeight": "70vh",
      "overflowX": "hidden",
      "width": "100vw",
    }
  }
>
  <div
    className="hidden lg:block"
    style={
      Object {
        "backgroundImage": "url([object Object])",
        "backgroundSize": "cover",
        "height": "70vh",
        "width": "100%",
      }
    }
  />
  <div
    className="text-right flex flex-col"
    data-aos="fade-left"
    style={
      Object {
        "height": "70vh",
        "width": "100%",
      }
    }
  >
    <div
      className="my-auto mx-auto flex flex-col sm:p-8 md:p-0"
    >
      <div
        className="grid grid-cols-2"
      >
        <img
          alt="Logo MWCollective"
          src="test-file-stub"
          style={
            Object {
              "width": "6rem",
            }
          }
        />
        <p
          className="inriasans-reg undefined my-auto"
        >
          studio projektowe MWCollective
        </p>
      </div>
      <h2
        className="mont-black my-5 undefined"
        style={
          Object {
            "fontSize": "3rem",
          }
        }
      >
        mniej znaczy wiÄ™cej
      </h2>
      <p
        className="inriasans-reg my-5 pb-5 undefined"
        style={
          Object {
            "width": "20rem",
          }
        }
      >
        testowy opis
      </p>
    </div>
  </div>
</header>
`;
